test/com/amazon/ion/impl/bin/IonManagedBinaryWriterTest.java

```diff
@@ -149,17 +211,48 @@ public void testFlushImmediatelyAfterIVM() throws Exception
 
     @Test
     public void testAutoFlush() throws Exception{
-
         IonReader reader = system().newReader(source.toByteArray());
         ByteArrayOutputStream actual = new ByteArrayOutputStream();
-        IonWriter actualWriter = IonBinaryWriterBuilder.standard().withAutoFlushEnbaled(true).build(actual);
+        IonWriter actualWriter = IonBinaryWriterBuilder.standard().withAutoFlushEnabled(true).build(actual);
         while (reader.next() != null) {
             actualWriter.writeValue(reader);
         }
-        actualWriter.finish();
+        actualWriter.close();
         if (lstAppendMode.isEnabled() && autoFlushMode.isEnabled()) {
             assertArrayEquals(actual.toByteArray(), expectedOut.toByteArray());
         }
+        assertArrayEquals(actual.toByteArray(), expectedOut.toByteArray());
```

Comment:
This is the same assertion from line 222. This should fail if auto flush is disabled, but I see that we're not actually running this test with auto-flush disabled. Consider using `@ParameterizedTest` to pass in both true and false for the value of `withAutoFlushEnabled`, then change this to an assertion of data model equality (i.e. using Equivalence.ionEquals). That way we can assert different behavior based on whether auto-flush is enabled, but verify in both cases that the streams are data model equivalent, which is what ultimately matters to the user.